jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java

```diff
@@ -64,6 +64,7 @@
     private boolean _delayDispatchUntilContent = true;
     private boolean _persistentConnectionsEnabled = true;
     private int _maxErrorDispatches = 10;
+    private boolean _useDirectByteBuffers = false;
```

Comment:
I think a single boolean is too coarse a switch.   Direct buffers may be beneficial for request, but not response headers/chunks of vice versa.

Perhaps an enum set to say which types of buffers are direct or not.

Also, currently ChannelEndPoint just returns true for isOptimizedForDirectBuffers, it may be best to have to influenced by this kind of configuration also.